virtualization/windowscontainers/container-networking/architecture.md

```diff
@@ -0,0 +1,67 @@
+> ## Unsupported features and network options 
+> The following networking options are currently **NOT** supported in Windows:
+>
+> | Command        | Unsupported Option   |
+> | ---------------|:--------------------:|
+> | ``docker run``|   ``--ip6``, ``--dns-option`` |
+> | ``docker network create``| ``--aux-address``, ``--internal``, ``--ip-range``, ``--ipam-driver``, ``--ipam-opt``, ``--ipv6``, ``--opt encrypted`` |
+
+
+## Basic networking architecture
+This topic provides an overview of how Docker creates and manages networks on Windows. Windows containers function similarly to virtual machines in regards to networking. Each container has a virtual network adapter (vNIC) which is connected to a Hyper-V virtual switch (vSwitch). Windows supports five different networking drivers or modes which can be created through Docker: *nat*, *overlay*, *transparent*, *l2bridge*, and *l2tunnel*. Depending on your physical network infrastructure and single- vs multi-host networking requirements, you should choose the network driver which best suits your needs.
+
+
+![text](media/windowsnetworkstack-simple.png)
+
+
+The first time the docker engine runs, it will create a default NAT network, 'nat', which uses an internal vSwitch and a Windows component named `WinNAT`. If there are any pre-existing external vSwitches on the host which were created through PowerShell or Hyper-V Manager, they will also be available to Docker using the *transparent* network driver and can be seen when you run the ``docker network ls`` command.  
+
+
+![text](media/docker-network-ls.png)
+
+
+> - An ***internal*** vSwitch is one which is not directly connected to a network adapter on the container host 
+
+> - An ***external*** vSwitch is one which _is_ directly connected to a network adapter on the container host  
+
+
+![text](media/get-vmswitch.png)
+
+
+The 'nat' network is the default network for containers running on Windows. Any containers that are run on Windows without any flags or arguments to implement specific network configurations will be attached to the default 'nat' network, and automatically assigned an IP address from the 'nat' network's internal prefix IP range. The default internal IP prefix used for 'nat' is 172.16.0.0/16. 
+
+
+## Container Network Management with Host Network Service
+
+The Host Networking Service (HNS) and the Host Compute Service (HCS) work together to create containers and attach endpoints to a network.
+
+### Network Creation
+  - HNS creates a Hyper-V virtual switch for each network
+  - HNS creates NAT and IP pools as required
+
+### Endpoint Creation
+  - HCS creates network compartment per container endpoint
```

Comment:
HNS creates "network namespace". Replace all instances of compartment with "namespace"